core/src/test/java/io/neow3j/protocol/core/OfflineServiceTest.java

```diff
@@ -0,0 +1,111 @@
+                () -> service.send(new Request(), NeoGetBlock.class));
+        assertThat(thrown.getMessage(), containsString(EXPECTED_OFFLINESERVICE_EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    public void testSendAsync() {
+        OfflineServiceException thrown = assertThrows(OfflineServiceException.class,
+                () -> service.sendAsync(new Request(), NeoGetBlock.class));
+        assertThat(thrown.getMessage(), containsString(EXPECTED_OFFLINESERVICE_EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    public void testSubscribeWithOnError() {
+        OfflineServiceException thrown = assertThrows(OfflineServiceException.class,
+                () -> service.subscribe(new Request(), "", Notification.class));
+        assertThat(thrown.getMessage(), containsString(EXPECTED_OFFLINESERVICE_EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    public void testClose() {
+        OfflineServiceException thrown = assertThrows(OfflineServiceException.class, service::close);
+        assertThat(thrown.getMessage(), containsString(EXPECTED_OFFLINESERVICE_EXCEPTION_MESSAGE));
+    }
+
+    // Tests with a Neow3j instance that has an OfflineService
+
+    @Test
+    public void testNeow3jSend() {
+        OfflineServiceException thrown = assertThrows(OfflineServiceException.class,
+                () -> neow3j.getBestBlockHash().send());
+        assertThat(thrown.getMessage(), containsString(EXPECTED_OFFLINESERVICE_EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    public void testNeow3jSendAsync() {
+        OfflineServiceException thrown = assertThrows(OfflineServiceException.class,
+                () -> neow3j.getBestBlockHash().sendAsync());
+        assertThat(thrown.getMessage(), containsString(EXPECTED_OFFLINESERVICE_EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    public void testNeow3jSubscribeWithOnError() throws IOException, InterruptedException {
+        AtomicBoolean onErrorExecuted = new AtomicBoolean(false);
+        neow3j.subscribeToNewBlocksObservable(true).subscribe(NeoGetBlock::getBlock, (e) -> onErrorExecuted.set(true));
+        Thread.sleep(100);
+        assertTrue(onErrorExecuted.get());
+    }
+
+    @Test
+    public void testNeow3jCatchupAndSubscribeWithOnError() throws InterruptedException {
+        AtomicBoolean onErrorExecuted = new AtomicBoolean(false);
+        neow3j.catchUpToLatestAndSubscribeToNewBlocksObservable(BigInteger.ZERO, true)
+                .subscribe(NeoGetBlock::getBlock, (e) -> onErrorExecuted.set(true));
+        Thread.sleep(100);
+        assertTrue(onErrorExecuted.get());
+    }
+
+    @Test
+    public void testNeow3jReplayBlocksObservableSubscribe() throws InterruptedException {
+        AtomicBoolean onErrorExecuted = new AtomicBoolean(false);
+        neow3j.replayBlocksObservable(BigInteger.ZERO, BigInteger.ZERO, true)
+                .subscribe(NeoGetBlock::getBlock, (e) -> onErrorExecuted.set(true));
+        Thread.sleep(100);
```

Comment:
and here.

Reply:
See https://github.com/neow3j/neow3j/pull/1102#discussion_r2037624918.